diff --git a/src/utils1.c b/src/utils1.c
index b8dedd5..c5ec75a 100644
--- a/src/utils1.c
+++ b/src/utils1.c
@@ -1104,7 +1104,7 @@ size_t  bufsize = 100;
 /*---------------------------------------------------------------------*
  *                           Timing procs                              *
  *---------------------------------------------------------------------*/
-#if !defined(_WIN32) && !defined(__Fuchsia__)
+#if !defined(_WIN32) && !defined(__Fuchsia__) && !defined(PSP)
 
 #include <sys/time.h>
 #include <sys/resource.h>
@@ -1198,7 +1198,7 @@ struct timeval tv;
     if (usec) *usec = (l_int32)tv.tv_usec;
 }
 
-#elif defined(__Fuchsia__) /* resource.h not implemented on Fuchsia. */
+#elif defined(__Fuchsia__) || defined(PSP) /* resource.h not implemented on Fuchsia. */
 
     /* Timer functions are used for testing and debugging, and
      * are stubbed out.  If they are needed in the future, they
