From ba40bc86ee1e4f3e04447239ac1c7fbda769cc21 Mon Sep 17 00:00:00 2001
From: Wouter Wijsman <wwijsman@live.nl>
Date: Wed, 30 Jul 2025 21:06:13 +0200
Subject: [PATCH] psp: fix audio not playing

---
 src/audio/psp/SDL_pspaudio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/audio/psp/SDL_pspaudio.c b/src/audio/psp/SDL_pspaudio.c
index 18546e88fc7ac..1b095e97f2fd2 100644
--- a/src/audio/psp/SDL_pspaudio.c
+++ b/src/audio/psp/SDL_pspaudio.c
@@ -114,7 +114,7 @@ static bool PSPAUDIO_PlayDevice(SDL_AudioDevice *device, const Uint8 *buffer, in
     } else {
         rc = sceAudioOutputPannedBlocking(device->hidden->channel, PSP_AUDIO_VOLUME_MAX, PSP_AUDIO_VOLUME_MAX, (void *) buffer);
     }
-    return (rc == 0);
+    return (rc >= 0);
 }
 
 static bool PSPAUDIO_WaitDevice(SDL_AudioDevice *device)
