diff --git a/CMakeLists.txt b/CMakeLists.txt
index e22f2cf..b739707 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -117,7 +117,9 @@ if (NOT MSVC)
   find_package(Threads)
   if (CMAKE_USE_PTHREADS_INIT)
     add_definitions("-DHAVE_PTHREAD")
-    list(APPEND THIRD_PARTY_LIBS Threads::Threads)
+    if (NOT PSP)
+      list(APPEND THIRD_PARTY_LIBS Threads::Threads)
+    endif()
     list(APPEND PC_LIBS_PRIV -pthread)
   endif ()
 endif ()
@@ -502,7 +504,7 @@ if (HB_BUILD_SUBSET)
   endif ()
 endif ()
 
-if (UNIX OR MINGW OR VITA)
+if (UNIX OR MINGW OR VITA OR PSP)
   # Make symbols link locally
   include (CheckCXXCompilerFlag)
   CHECK_CXX_COMPILER_FLAG(-Bsymbolic-functions CXX_SUPPORTS_FLAG_BSYMB_FUNCS)
