diff --git a/CMakeLists.txt b/CMakeLists.txt
index e22f2cf..b739707 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -116,11 +116,17 @@ if (NOT MSVC)
   find_package(Threads)
   if (CMAKE_USE_PTHREADS_INIT)
     add_definitions("-DHAVE_PTHREAD")
-    list(APPEND THIRD_PARTY_LIBS Threads::Threads)
+    if (NOT PSP)
+      list(APPEND THIRD_PARTY_LIBS Threads::Threads)
+    endif()
     list(APPEND PC_LIBS_PRIV -pthread)
   endif ()
 endif ()
 
+if (PSP)
+  list(APPEND PC_LIBS_PRIV -lstdc++)
+endif ()
+
 if (MSVC)
   add_definitions(-wd4244 -wd4267 -D_CRT_SECURE_NO_WARNINGS -D_CRT_NONSTDC_NO_WARNINGS)
 endif ()
