diff -ruN ../orig/curl-7.64.1/CMake/FindPolarSSL.cmake curl-7.64.1/CMake/FindPolarSSL.cmake
--- ../orig/curl-7.64.1/CMake/FindPolarSSL.cmake	1969-12-31 21:00:00
+++ curl-7.64.1/CMake/FindPolarSSL.cmake	2023-03-15 10:57:18
@@ -0,0 +1,11 @@
+find_path(POLARSSL_INCLUDE_DIRS polarssl/ssl.h)
+
+find_library(POLARSSL_LIBRARY polarssl)
+
+set(POLARSSL_LIBRARIES "${POLARSSL_LIBRARY}")
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(PolarSSL DEFAULT_MSG
+    POLARSSL_INCLUDE_DIRS POLARSSL_LIBRARY)
+
+mark_as_advanced(POLARSSL_INCLUDE_DIRS POLARSSL_LIBRARY)
diff -ruN ../orig/curl-7.64.1/CMakeLists.txt curl-7.64.1/CMakeLists.txt
--- ../orig/curl-7.64.1/CMakeLists.txt	2019-03-25 05:42:48
+++ curl-7.64.1/CMakeLists.txt	2023-03-15 10:56:43
@@ -301,9 +301,10 @@
     CMAKE_USE_WINSSL OFF)
 endif()
 option(CMAKE_USE_MBEDTLS "Enable mbedTLS for SSL/TLS" OFF)
+option(CMAKE_USE_POLARSSL "Enable PolarSSL for SSL/TLS" OFF)
 
 set(openssl_default ON)
-if(WIN32 OR CMAKE_USE_DARWINSSL OR CMAKE_USE_WINSSL OR CMAKE_USE_MBEDTLS)
+if(WIN32 OR CMAKE_USE_DARWINSSL OR CMAKE_USE_WINSSL OR CMAKE_USE_MBEDTLS OR CMAKE_USE_POLARSSL)
   set(openssl_default OFF)
 endif()
 option(CMAKE_USE_OPENSSL "Use OpenSSL code. Experimental" ${openssl_default})
@@ -313,6 +314,7 @@
   CMAKE_USE_DARWINSSL
   CMAKE_USE_OPENSSL
   CMAKE_USE_MBEDTLS
+  CMAKE_USE_POLARSSL
 )
 if(enabled_ssl_options_count GREATER "1")
   set(CURL_WITH_MULTI_SSL ON)
@@ -375,6 +377,14 @@
   check_symbol_exists(RAND_egd    "${CURL_INCLUDES}" HAVE_RAND_EGD)
 endif()
 
+if(CMAKE_USE_POLARSSL)
+  find_package(PolarSSL REQUIRED)
+  set(SSL_ENABLED ON)
+  set(USE_POLARSSL ON)
+  list(APPEND CURL_LIBS ${POLARSSL_LIBRARIES})
+  include_directories(${POLARSSL_INCLUDE_DIRS})
+endif()
+
 if(CMAKE_USE_MBEDTLS)
   find_package(MbedTLS REQUIRED)
   set(SSL_ENABLED ON)
@@ -705,7 +715,7 @@
   endif()
 endif()
 
-if(CURL_CA_PATH_SET AND NOT USE_OPENSSL AND NOT USE_MBEDTLS)
+if(CURL_CA_PATH_SET AND NOT USE_OPENSSL AND NOT USE_MBEDTLS AND NOT USE_POLARSSL)
   message(FATAL_ERROR
           "CA path only supported by OpenSSL, GnuTLS or mbed TLS. "
           "Set CURL_CA_PATH=none or enable one of those TLS backends.")
@@ -1182,6 +1192,7 @@
 _add_if("WinSSL"        SSL_ENABLED AND USE_WINDOWS_SSPI)
 _add_if("OpenSSL"       SSL_ENABLED AND USE_OPENSSL)
 _add_if("DarwinSSL"     SSL_ENABLED AND USE_DARWINSSL)
+_add_if("PolarSSL"      SSL_ENABLED AND USE_POLARSSL)
 _add_if("mbedTLS"       SSL_ENABLED AND USE_MBEDTLS)
 _add_if("IPv6"          ENABLE_IPV6)
 _add_if("unix-sockets"  USE_UNIX_SOCKETS)
@@ -1200,7 +1211,7 @@
                         (HAVE_GSSAPI OR USE_WINDOWS_SSPI))
 # NTLM support requires crypto function adaptions from various SSL libs
 # TODO alternative SSL libs tests for SSP1, GNUTLS, NSS
-if(NOT CURL_DISABLE_CRYPTO_AUTH AND (USE_OPENSSL OR USE_WINDOWS_SSPI OR USE_DARWINSSL OR USE_MBEDTLS))
+if(NOT CURL_DISABLE_CRYPTO_AUTH AND (USE_OPENSSL OR USE_WINDOWS_SSPI OR USE_DARWINSSL OR USE_MBEDTLS OR USE_POLARSSL))
   _add_if("NTLM"        1)
   # TODO missing option (autoconf: --enable-ntlm-wb)
   _add_if("NTLM_WB"     NOT CURL_DISABLE_HTTP AND NTLM_WB_ENABLED)
diff -ruN ../orig/curl-7.64.1/include/curl/curl.h curl-7.64.1/include/curl/curl.h
--- ../orig/curl-7.64.1/include/curl/curl.h	2019-03-25 05:42:50
+++ curl-7.64.1/include/curl/curl.h	2023-03-15 10:56:26
@@ -51,7 +51,7 @@
 
 #if defined(__FreeBSD__) && (__FreeBSD__ >= 2)
 /* Needed for __FreeBSD_version symbol definition */
-#include <osreldate.h>
+//#include <osreldate.h>
 #endif
 
 /* The include stuff here below is mainly for time_t! */
