diff -ruN ../orig/curl-7.64.1/include/curl/curl.h curl-7.64.1/include/curl/curl.h
--- ../orig/curl-7.64.1/include/curl/curl.h	2019-03-25 05:42:50
+++ curl-7.64.1/include/curl/curl.h	2023-03-15 10:56:26
@@ -51,7 +51,7 @@
 
 #if defined(__FreeBSD__) && (__FreeBSD__ >= 2)
 /* Needed for __FreeBSD_version symbol definition */
-#include <osreldate.h>
+//#include <osreldate.h>
 #endif
 
 /* The include stuff here below is mainly for time_t! */
